README.md
guidaemacsauctex
================

Contiene i sorgenti della guida tematica Guida pratica all’uso di GNU Emacs &amp; AUCTeX

ChangeLog
2013-04-04  Orlando Iovino  <orlando [dot] iovino [at] yahoo [dot] it>

	* guidaemacsauctex.tex: modificato il formato della versione.



dotemacs/windows/init.el
;;; init.el --- Personal GNU Emacs configuration
;;
;; Copyright (c) 2013 Orlando Iovino
;;
;; Author: Orlando Iovino

;; This file is NOT part of GNU Emacs.

;;; Commentary:

;; Generic customizations.

;;; License:

;; This program is free software; you can redistribute it and/or modify it under
;; the terms of the GNU General Public License as published by the Free Software
;; Foundation; either version 3 of the License, or (at your option) any later
;; version.
;;
;; This program is distributed in the hope that it will be useful, but WITHOUT
;; ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
;; FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
;; details.
;;
;; You should have received a copy of the GNU General Public License along with
;; GNU Emacs; see the file COPYING. If not, write to the Free Software
;; Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
;; USA.

;; Code:

(add-hook 'LaTeX-mode-hook 'TeX-PDF-mode); ...... attiva di default la pdflatex-mode
(add-hook 'LaTeX-mode-hook 'LaTeX-math-mode); ... attiva di default la math-mode
(add-hook 'LaTeX-mode-hook 'turn-on-reftex); .... attiva di default la reftex-mode

;; Chiede qual e' il master file all'apertura di Emacs
(setq-default TeX-master nil)

;; Crea una directory con informazioni sui pacchetti
(setq TeX-parse-self t)
(setq TeX-auto-save t)

;; Visualizza bad-boxes/warning
(setq TeX-debug-bad-boxes t)
(setq TeX-debug-warnings t)

;; Impostazioni per il correttore ortografico
(setq-default ispell-program-name "aspell")
(setq-default ispell-extra-args '("--reverse"))
(setq ispell-dictionary "italiano")

;; Ricerca diretta inversa
(setq TeX-source-correlate-method (quote synctex))
(setq TeX-source-correlate-mode t)
(setq TeX-source-correlate-start-server t)

(setq TeX-view-program-list
      '(("Sumatra PDF" ("\"C:/Program Files (x86)/SumatraPDF/SumatraPDF.exe\" -reuse-instance" (mode-io-correlate " -forward-search %b %n") " %o"))))

(setq TeX-view-program-selection
      '(((output-dvi style-pstricks) "dvips and start")
	(output-dvi "Yap")
	(output-pdf "Sumatra PDF")
	(output-html "start")))

;; Coppia di graffe automatica per _ e ^
(setq TeX-electric-sub-and-superscript t)

;; Didascalia tabelle sopra
(setq LaTeX-top-caption-list (quote ("table")))

;; A capo automatico
(add-hook 'TeX-mode-hook 'turn-on-auto-fill)

;;; init.el ends here


style/guidatematica.el
;;; guidatematica.el --- AUCTeX style for `guidatematica.cls'

;; Copyright (C) 2013 Orlando Iovino

;; Author: Orlando Iovino <orlando[dot]iovino[at]yahoo[dot]it>
;; Created: 2013-02-07
;; Keywords: tex

;; This file is not part of AUCTeX.

;; AUCTeX is free software; you can redistribute it and/or modify it
;; under the terms of the GNU General Public License as published by
;; the Free Software Foundation; either version 3, or (at your option)
;; any later version.

;; AUCTeX is distributed in the hope that it will be useful, but
;; WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
;; General Public License for more details.

;; You should have received a copy of the GNU General Public License
;; along with AUCTeX; see the file COPYING. If not, write to the Free
;; Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
;; 02110-1301, USA.

;;; Commentary:

;; This file adds support for the `guidatematica.cls'.

;;; Code:

(TeX-add-style-hook "guidatematica"
 (lambda ()
   (TeX-add-symbols
    ;; Comandi
    '("ProvidesFile" "Nome del file" [ "aaaa/mm/gg v.x.y Descrizione" ])
    '("meta" "Argomento")
    '("marg" "Argomento")
    '("oarg" "Argomento")
    '("garg" "x,y")
    '("comando" "Comando (con \\)")
    '("cs" "Comando (senza \\)")
    '("Bambiente" "Nome dell'ambiente")
    '("Eambiente" "Nome dell'ambiente")
    '("Sambiente" "Nome dell'ambiente" "Numero di argomenti")
    '("Dambiente" "Nome dell'ambiente" "Numero di argomenti" "Arg. default")
    '("Arg" "Argomento")
    '("GetFileInfo" "Nome del file")
    '("setcopymark" "Testo del copymark")
    '("subtitle" "Sottotitolo")
    '("showindexmarks")
    '("hideindexmarks")
    '("texttm" "Argomento")
    '("texttv" "Argomento")
    '("tvfamily")
    '("textl" "Numero")
    '("prebibhook" "Testo")
    '("ambstyle" "Nome dell'ambiente")
    '("classstyle" "Nome della classe")
    '("filestyle" "Nome del file")
    '("packstyle" "Nome del pacchetto")
    '("progstyle" "Nome del programma")
    '("opzstyle" "Nome dell'opzione")
    '("contastyle" "Nome del contatore")
    '("stilestyle" "Nome dello stile di pagina")
    '("numeristyle" "Nome dello stile di numerazione")
    '("umisurastyle" "Unità di misura")
    '("chiavestyle" "Nome della chiave")
    '("descrittorestyle" "Codice del descrittore")
    '("posizionestyle" "Codice di posizionamento")
    '("allineamentostyle" "Codice di allineamento")
    ;; Comandi collegati all'indice analitico
    '("opz" "Nome dell'opzione")
    '("conta" "Nome del contatore")
    '("stile" "Nome dello stile di pagina")
    '("numeri" "Nome dello stile di numerazione")
    '("umisura" "Unità di misura")
    '("chiave" "Nome della chiave")
    '("descrittore" "Codice del descrittore")
    '("posizione" "Codice di posizionamento")
    '("allineamento" "Codice di allineamento")
    '("amb" "Nome dell'ambiente")
    '("class" "Nome della classe")
    '("file" "Nome del file")
    '("pack" "Nome del pacchetto")
    '("prog" "Nome del programma"))
   ;; Ambienti
   (LaTeX-add-environments
    '("blockdescription" LaTeX-env-item) ;; MODIFICARE
    '("plaindescription" LaTeX-env-item) ;; MODIFICARE
    '("compactitemize" LaTeX-env-item)
    '("compactenumerate" LaTeX-env-item)
    '("compactdescription" LaTeX-env-item) ;; MODIFICARE
    '("sintassi"))

   (TeX-run-style-hooks
    "memoir"
    "memoir10")
   (if (member "ipertesto" TeX-active-styles)
       (TeX-run-style-hooks
	"hyperref"))
   (if (eq TeX-engine 'xetex)
       (TeX-run-style-hooks
	"fontspec"
	"unicode-math"
	"polyglossia")
     (TeX-run-style-hooks
      "utf8"
      "inputenc"
      "T1"
      "fontenc"
      "italian"
      "babel"
      "cfr-lm"))
   (TeX-run-style-hooks
    "graphicx"
    "metalogo"
    "pict2e"
    "microtype"
    "etoolbox"
    "xcolor"
    "natbib"
    "multicol"
    "imakeidx"
    "fancyvrb"
    "afterpage"
    "enumitem"
    "listings"
    "canoniclayout")

   ;; Fontification
   (when (and (featurep 'font-latex)
	      (eq TeX-install-font-lock 'font-latex-setup))
     (font-latex-add-keywords '(;; Comandi
				("ProvidesFile" "{[")
				("meta" "{")
				("marg" "{")
				("oarg" "{")
				("garg" "{")
				("comando" "{")
    				("cs" "{")
				("Bambiente" "{")
				("Eambiente" "{")
				("Sambiente" "{{")
				("Dambiente" "{{{")
				("Arg" "{")
				("GetFileInfo" "{")
				("setcopymark" "{")
				("subtitle" "{")
				("showindexmarks")
				("hideindexmarks")
				("texttm" "{")
				("texttv" "{")
				("tvfamily")
				("textl" "{")
				("prebibhook" "{")
				("ambstyle" "{")
				("classstyle" "{")
				("filestyle" "{")
				("packstyle" "{")
				("progstyle" "{")
				("opzstyle" "{")
				("contastyle" "{")
				("stilestyle" "{")
				("numeristyle" "{")
				("umisurastyle" "{")
				("chiavestyle" "{")
				("descrittorestyle" "{")
				("posizionestyle" "{")
				("allineamentostyle" "{")
				;; Comandi collegati all'indice analitico
				("opz" "{")
				("conta" "{")
				("stile" "{")
				("numeri" "{")
				("umisura" "{")
				("chiave" "{")
				("descrittore" "{")
				("posizione" "{")
				("allineamento" "{")
				("amb" "{")
				("class" "{")
				("file" "{")
				("pack" "{")
				("prog" "{"))
			      'function))))

(defvar LaTeX-guidatematica-class-options '("ipertesto")
  "Class options for the guidatematica class.")

;;; guidatematica.el ends here
